src/componentcard/card.js

```diff
@@ -0,0 +1,43 @@
+import React, {useState, useEffect} from 'react';
+import './card.css';
+import axios from "axios";
+
+function Card({ url }) {
+    const [singlePokemon, setSinglePokemon] = useState({});
+    const [moves, setMoves] = useState([]);
+    const [abilities, setAbilities] = useState([]);
+    const [ability, setAbility] = useState({});
+    useEffect(() => {
+        async function getSinglePokemon() {
+            try {
+                const { data: {name, weight, sprites: {front_default}, moves, abilities, }} = await axios.get(url);
+                    setSinglePokemon({
+                        name: name,
+                        weight: weight,
+                        image: front_default,
+                    });
+                    setAbilities(abilities.length);
+                    setMoves(moves.length);
+                    setAbility(ability);
```

Comment:
Het is niet heel erg om meerdere useStates te gebruiken, maar als je ze nooit los van elkaar zal updaten, dan kun je er net zo goed één useState voor gebruiken.